deps/lib_julia/ViZDoomJuliaModule.cpp

```diff
@@ -312,16 +286,21 @@ JLCXX_MODULE ViZDoom(jlcxx::Module &mod)
         .method("object_velocity_y", [](Label &l) { return l.objectVelocityY; })
         .method("object_velocity_z", [](Label &l) { return l.objectVelocityZ; });
 
-    mod.add_type<Buffer>("Buffer");
+    // This line breaks CxxWrap, double registration, already mapped type, etc...
+    // mod.add_type<Buffer>("Buffer");
+
     mod.add_type<GameState>("GameState")
         .method("number", [](GameState &gs) { return gs.number; })
         .method("tic", [](GameState &gs) { return gs.tic; })
-        .method("game_variables", [](GameState &gs) { return jlcxx::ArrayRef<double, 1>(&(gs.gameVariables[0]), gs.gameVariables.size()); })
+        .method("game_variables", [](GameState &gs) { return jlcxx::ArrayRef<double, 1>(gs.gameVariables.data(), gs.gameVariables.size()); })
         .method("screen_buffer", [](GameState &gs) { return gs.screenBuffer; })
         .method("depth_buffer", [](GameState &gs) { return gs.depthBuffer; })
         .method("labels_buffer", [](GameState &gs) { return gs.labelsBuffer; })
-        .method("automap_buffer", [](GameState &gs) { return gs.automapBuffer; })
-        .method("labels", [](GameState &gs) { return jlcxx::ArrayRef<Label, 1>(&(gs.labels[0]), gs.labels.size()); });
+        .method("automap_buffer", [](GameState &gs) { return gs.automapBuffer; });
+        // TODO: figure out how to return this? The below code breaks
+        // .method("labels", [](GameState &gs) { 
+        //     return jlcxx::ArrayRef<Label, 1>(labels.data(), labels.size()); 

```

Comment:
https://github.com/JuliaInterop/CxxWrap.jl#reference-native-julia-arrays

Maybe give it a try to `*Label` ?